taichi/ir/frontend_ir.cpp

```diff
@@ -667,11 +696,19 @@ void IndexExpression::type_check(CompileConfig *) {
   for (int i = 0; i < indices.exprs.size(); i++) {
```

Comment:
Dimension compatibility checks (before this line) should also be aware of this special case.